src/Test/QuickCheck/Arbitrary.purs

```diff
@@ -143,3 +146,56 @@ instance arbitraryLazy :: Arbitrary a => Arbitrary (Lazy a) where
 
 instance coarbLazy :: Coarbitrary a => Coarbitrary (Lazy a) where
   coarbitrary a = coarbitrary (force a)
+
+
+instance arbitraryNoArguments :: Arbitrary NoArguments where
+  arbitrary = pure NoArguments
+
+instance coarbitraryNoArguments :: Coarbitrary NoArguments where
+  coarbitrary NoArguments = id
+
+instance arbitrarySum :: (Arbitrary l, Arbitrary r) => Arbitrary (Sum l r) where
+  arbitrary = arbitrary >>= if _ then Inl <$> arbitrary else Inr <$> arbitrary
```

Comment:
One slight problem here is that this will not be evenly distributed if there are more than two constructors. The first would get chosen with 50% probability.